docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,190 @@
+
+.Building an optimized server image
+<@kc.build parameters="--db=postgres --db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+.Starting the server
+<@kc.start/>
+
+As a rule of thumb, prefer building a server image to achieve the optimal runtime as well as running the `start` command without any parameter for faster startup time.
+
+=== Overriding settings
+
+Once you have built a server image using the `build` command, you should be able to override any property that is available from the `start` command.
+
+For instance, if you have built an image as follows:
+
+<@kc.build parameters="--db=postgres --db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+You should be able to override the `db-username`, `hostname`, or any other property available to the `start` command:
+
+<@kc.start parameters="--db-username foo --hostname newkeycloak.acme.com"/>
+
+As a result, the server should start using all the configuration options previously set using the `build` command but overriding those that you explicitly
+set when starting the server.
+
+This configuration approach can be useful when building an opinionated server image with the possibility to override specific options that are allowed to
+change when starting the server (those available to the `start` command).
+
+== Using placeholders
+
+You should be able to use placeholders to dynamically change the value of configuration options when starting the server.
+
+For that, you should use the following format when defining the value of an option:
+
+```
+db.username=${r"${my.db.user}"}
+```
+
+By using a placeholder like in the example above, the server is going to try to resolve the `my.db.user` variable from:
+
+* A system property using the same key
+* An environment variable by replacing dots with `_`.
+
+Placeholders can also be used when building a server image using the `build` command so that the value of a configuration option is updated
+with the value set to the placeholder (system property or environment variable) when starting the server.
+
+[NOTE]
+You should not use placeholders for configuration options that can only be set through the `build` command. The reason is that these options
+can not change once they are persisted into the server image. Prefer using placeholders only for the configurations options available to
+the `start` command.
+
+== Configuration profiles
+
+Most of the time the server is deployed to different environments like testing, staging, and production environments. Although they share some
+core configuration options (e.g: database, clustering, etc) some other options might change on a per-environment basis.
+In order to make easier to support multiple configurations, the server allows you to create configuration profiles.
```

Comment:
In order to make _it_ easier to..., _Keycloak_ allows... ?

Reply:
If I have to change `server` then I would need to change a lot of other places. The reason why I'm not using `Keycloak` is to be more neutral about community/product. We should probably have a macro to resolve the proper name similarly as how we do in the current documentation?

Reply:
works for me :) but that macro is a good idea, in general.